cirq-core/cirq/vis/circuit_to_latex_quantikz.py

```diff
@@ -0,0 +1,710 @@
+            is_symbolic_or_special = any(
+                char.isalpha()
+                for char in s_exponent
+                if char.lower() not in ["e"]  # Exclude 'e' for scientific notation
+            )
+            if not is_symbolic_or_special:  # If it looks like a number
+                try:
+                    py_float = float(sympy.N(exponent))  # type: ignore
+                    # If the sympy evaluated float is an integer value
+                    if py_float.is_integer():
+                        exp_str = str(int(py_float))
+                    else:
+                        # Format to specified precision for rounding
+                        rounded_str = format(py_float, float_format_string)
+                        # Convert back to float and then to string to remove unnecessary trailing zeros
+                        exp_str = str(float(rounded_str))
+                except (TypeError, ValueError, AttributeError, sympy.SympifyError):  # type: ignore
+                    exp_str = s_exponent  # Fallback to Sympy's string representation
+            else:  # Symbolic expression
+                exp_str = s_exponent
+        else:  # For other types (int, strings not sympy objects)
+            exp_str = str(exponent)
+
+        # LaTeX replacements for pi
+        exp_str = exp_str.replace("pi", r"\pi").replace("π", r"\pi")
+
+        # Handle underscores: replace "_" with "\_" if not part of a LaTeX command
+        if "_" in exp_str and "\\" not in exp_str:
+            exp_str = exp_str.replace("_", r"\_")
+        return exp_str
+
+    def _get_gate_name(self, gate: "cirq.Gate") -> str:
+        """Determines the appropriate LaTeX string for a given Cirq gate.
+
+        This method attempts to derive a suitable LaTeX name for the gate,
+        considering its type, whether it's a power gate, and if parameters
+        should be displayed. It uses internal mappings and `cirq.circuit_diagram_info`.
+
+        Args:
+            gate: The `cirq.Gate` object to name.
+
+        Returns:
+            A string representing the LaTeX name of the gate (e.g., "H",
+            "Rx(0.5)", "CZ").
+        """
+        gate_type = type(gate)
+        if gate_type.__name__ == "ThermalChannel":
+            return "\\Lambda_\\mathrm{th}"
+        if (simple_name := self._SIMPLE_GATE_MAP.get(gate_type)) is not None:
+            return simple_name
+
+        base_key = self._EXPONENT_GATE_MAP.get(gate_type)
+        if base_key is not None and hasattr(gate, "exponent") and gate.exponent == 1:
+            return self.current_gate_name_map.get(base_key, base_key)
+
+        if (param_base_key := self._PARAMETERIZED_GATE_BASE_NAMES.get(gate_type)) is not None:
+            mapped_name = self.current_gate_name_map.get(param_base_key, param_base_key)
+            if not self.show_parameters:
+                return mapped_name
+            try:
+                info_src = cirq.protocols if hasattr(cirq, "protocols") else cirq
```

Comment:
What's up with this?  This seems odd.  Why wouldn't cirq have the protocols package?  
(Also, this needs to change to "from cirq import protocols" above)

Same below.